server/src/server.rs

```diff
@@ -787,54 +781,95 @@ impl<HttpMiddleware, RpcMiddleware> Builder<HttpMiddleware, RpcMiddleware> {
 	/// # Examples
 	///
 	/// ```no_run
-	/// use hyper::service::{make_service_fn, service_fn};
-	/// use hyper::server::conn::AddrStream;
 	/// use jsonrpsee_server::{Methods, ServerHandle, ws, stop_channel};
 	/// use tower::Service;
 	/// use std::{error::Error as StdError, net::SocketAddr};
+	/// use futures_util::future::{self, Either};
+	/// use hyper_util::rt::{TokioIo, TokioExecutor};
 	///
-	/// fn run_server() -> ServerHandle {
-	///     let addr = SocketAddr::from(([127, 0, 0, 1], 0));
+	/// async fn run_server() -> ServerHandle {
+	///     let listener = tokio::net::TcpListener::bind(SocketAddr::from(([127, 0, 0, 1], 0))).await.unwrap();
```

Comment:
I guess this could move into the `tokio::spawn` and then `run_server` could be sync again (iirc this is in an example or two as well)?